### PR TITLE
Replace Manager's integrate(state) with integrate(state, finalTime)

### PR DESCRIPTION
**Pull request by carmichaelong**
Addresses #1462 by doing the following:
* Add a function `integrate(state, finalTime)`
* Deprecate functions related to `integrate(state)` without throwing warnings (e.g., member variables `_ti` and `_tf` cannot be deprecated without throwing a warning)
* Some clean up of the `integrate()` function

### CODE REVIEW

**Comment by chrisdembia** [comment]
The AppVeyor failures should go away if you merge with master.

File: OpenSim/Simulation/Manager/Manager.h [hunk 1]
```diff
@@ -180,9 +180,19 @@ class OSIMSIMULATION_API Manager
     void setIntegrator(SimTK::Integrator&);
 
     // Initial and final times
+    /** <b>(Deprecated)</b> Set the state's time using state.setTime(time). */
```
Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.h]: You can get doxygen to link to the correct SimTK method if you write something like `SimTK::State::getTime()`.

File: OpenSim/Simulation/Manager/Manager.h [hunk 2]
```diff
@@ -208,9 +218,11 @@ class OSIMSIMULATION_API Manager
     //--------------------------------------------------------------------------
     // EXECUTION
     //--------------------------------------------------------------------------
+    bool integrate(SimTK::State& s, double finalTime);
+    /** <b>(Deprecated)</b> Integrate to a specified finalTime using
+        Manager::integrate(state, finalTime). */
```
Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.h]: Can you check if doxygen generates a link for that method?

Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.h]: You might need to do `Manager::integrate(SimTK::State&, double)`.

File: OpenSim/Simulation/Manager/Manager.h [hunk 3]
```diff
@@ -208,9 +218,11 @@ class OSIMSIMULATION_API Manager
     //--------------------------------------------------------------------------
     // EXECUTION
     //--------------------------------------------------------------------------
+    bool integrate(SimTK::State& s, double finalTime);
```
Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.h]: Add a doxygen comment for this function, since we expect this to be the one people actually use.

Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.h]: Ah I see...it's in the cpp file. I think the comment from the cpp file should be moved here.

File: OpenSim/Simulation/Manager/Manager.cpp [hunk 4]
```diff
@@ -623,7 +623,131 @@ hasStateStorage() const
 //-----------------------------------------------------------------------------
 // INTEGRATION
 //-----------------------------------------------------------------------------
-///____________________________________________________________________________
+/**
+* Integrate the equations of motion for the specified model, given the current
+* state (at which the integration will start) and a finalTime. Make sure to
+* use state.setTime(initialTime) to specify a starting time before calling
```
Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.cpp]: Consider giving an example in this comment block, and also getting the link to `SimTK::State::setTime()` to work.

File: OpenSim/Simulation/Manager/Manager.cpp [hunk 5]
```diff
@@ -623,7 +623,131 @@ hasStateStorage() const
...
+* this function.
+*/
+bool Manager::
+integrate(SimTK::State& s, double finalTime)
+{
+    int step = 0; // for AnalysisSet::step()
+
+    if (!_integ) {
+        throw Exception("Manager::doIntegration(): "
```
Comment (chrisdembia) [inline OpenSim/Simulation/Manager/Manager.cpp]: Change message to `integrate()`?

**Comment by carmichaelong** [comment]
Updated the PR. Some notes:
* For some reason `SimTK::State::getTime()` and related methods aren't linking properly.
* The documentation does not say anything about intermixing the two `integrate()` calls. In theory, it should be fine since they funnel through the same code now, but separately one of them is deprecated so I wasn't sure if we needed more than that? I'm definitely open to ideas of how to make this clear to users.

**Comment by carmichaelong** [comment]
And oops, will do CHANGELOG update too. Forgot that in this round.

File: OpenSim/Simulation/SimbodyEngine/Test/testConstraints.cpp [hunk 6]
```diff
@@ -215,9 +215,8 @@ void integrateOpenSimModel(Model *osimModel, SimTK::State &osim_state)
     // In this case, the initial and final times are set based on
     // the range of times over which the controls are available.
     //Control *control;
-    manager.setInitialTime(0.0);
-    manager.setFinalTime(duration);
-    manager.integrate(osim_state);
+    osim_state.setTime(0.0);
+    manager.integrate(osim_state, duration);
```
Comment (aseth1) [inline OpenSim/Simulation/SimbodyEngine/Test/testConstraints.cpp]: :+1: cleaner and less ambiguous!

File: OpenSim/Simulation/Manager/Manager.cpp [hunk 7]
```diff
@@ -732,35 +712,49 @@ bool Manager::doIntegration(SimTK::State& s, int step) {
...
     s = _integ->getState();
 
     // CLEAR ANY INTERRUPT
     clearHalt();
 
     return true;
+
+}
+
+/**
```
Comment (aseth1) [inline OpenSim/Simulation/Manager/Manager.cpp]: No need for Doxygen comments in the .cpp

File: OpenSim/Simulation/Manager/Manager.h [hunk 8]
```diff
@@ -148,7 +148,7 @@ class OSIMSIMULATION_API Manager
      * must call setIntegrator() on your own. You should use one of the other
      * two constructors. */
     DEPRECATED_14("There will be no replacement for this constructor.")
```
Comment (aseth1) [inline OpenSim/Simulation/Manager/Manager.h]: Maybe indicate the use of `Manager(Model& )` or `Manager(Model&, Integrator&)` instead?

File: OpenSim/Simulation/Manager/Manager.h [hunk 9]
```diff
@@ -208,9 +220,37 @@ class OSIMSIMULATION_API Manager
...
+    * Example: Integrate from time = 0s to time = 10s, in 2s increments
+    * @code
+    * dTime = 2.0;
+    * finalTime = 10.0;
+    * int n = int(round(finalTime/dTime));
+    * state.setTime(0.0);
+    * for (int i = 1; i <= n; ++i) {
+    *     manager.integrate(state, i*dTime);
+    * }
+    * @endcode
```
Comment (aseth1) [inline OpenSim/Simulation/Manager/Manager.h]: :+1: Very clear usage. Thanks!

**Comment by chrisdembia** [comment]
LGTM. I look forward to using this cleaner interface.

**Pull request merged**